app/server/src/db/userStore.ts

```diff
@@ -98,6 +98,12 @@ export class UserStore {
         const amrString = await this._redis.hget(this._projectSampleKey(projectId, sampleId), "amr");
         return JSON.parse(amrString);
     }
+    async deleteSample(projectId: string, sampleHash: string) {
+        const sampleIds = await this._redis.smembers(this._projectSamplesKey(projectId));
+        const sampleId = sampleIds?.find((id) => id.startsWith(sampleHash));
```

Comment:
I don't think this quite works. In theory someone might have uploaded the same sample data twice with two different filenames (could happen!) - if they select the second one to delete, it shouldn't delete the first one, which I think is what would happen here. So I think the delete endpoint is going to have to take the filename as well as the sample hash unfortunately.